File: elixir/009/009.exs
defmodule Euler do
  def pythagorean_triple_with_sum(sum) when sum > 11 do
    pythagorean_triple_with_sum(sum, 2, 1)
  end

  defp pythagorean_triple_with_sum(sum, m, n) when m < sum and n < sum - 1 and m > n do
    a = (m * m) - (n * n)

    b = 2 * m * n

    c = (m * m) + (n * n)

    if (a + b + c == sum) && ((a * a) + (b * b) == (c * c)) do
      [a, b, c]
    else
      pythagorean_triple_with_sum(sum, m, n + 1)
    end
  end

  defp pythagorean_triple_with_sum(sum, m, n) when m < sum and n < sum - 1 do
    pythagorean_triple_with_sum(sum, m + 1, 1)
  end

  defp pythagorean_triple_with_sum(_, _, _) do
    []
  end
end

IO.puts Euler.pythagorean_triple_with_sum(1000)
|> Enum.reduce(1, &(&1 * &2))
|> to_string

File: javascript/009/009.js
"use strict";

const pythagoreanTripleWithSum = function(sum) {
  for (let m = 2; m < sum + 1; m += 1) {
    for (let n = 1; n < m; n += 1) {
      const a = (m * m) - (n * n);

      const b = 2 * m * n;

      const c = (m * m) + (n * n);

      if ((a + b + c === sum) && ((a * a) + (b * b) === (c * c))) {
        return [a, b, c];
      }
    }
  }

  return [];
};

console.log(pythagoreanTripleWithSum(1000).reduce((previousValue, currentValue) => previousValue * currentValue));
